com.unity.netcode.gameobjects/Runtime/Messaging/Messages/TimeSyncMessage.cs

```diff
@@ -0,0 +1,29 @@
+namespace Unity.Netcode
+{
+    internal struct TimeSyncMessage : INetworkMessage
+    {
+        public int Tick;
+
+        public void Serialize(ref FastBufferWriter writer)
+        {
+            writer.WriteValueSafe(this);
+        }
+
+        public static void Receive(ref FastBufferReader reader, NetworkContext context)
+        {
+            var networkManager = (NetworkManager)context.SystemOwner;
+            if (!networkManager.IsClient)
+            {
+                return;
+            }
+            reader.ReadValueSafe(out TimeSyncMessage message);
+            message.Handle(context.SenderId, networkManager);
+        }
```

Comment:
looking at this reminds me the old PerformanceRPC approach we dropped back in March in favor of RPC with ILPP.
this looks like PerformanceRPCs but called INetworkMessages with specific struct containers instead of NetworkBehaviours.
not to hate but now I kind of understand why we raised "but we have RPCs already, why INetworkMessages?" in the past.

Reply:
This serves a completely different use case than RPCs... you can't send a ConnectionRequest message as an RPC...

Reply:
yeah I know, I just figured why it seemed that way.